crates/ruff_linter/src/rules/ruff/rules/property_without_return.rs

```diff
@@ -0,0 +1,119 @@
+use ruff_macros::{ViolationMetadata, derive_message_formats};
+use ruff_python_ast::visitor::{Visitor, walk_expr, walk_stmt};
+use ruff_python_ast::{Expr, Stmt, StmtFunctionDef};
+use ruff_python_semantic::analyze::{function_type, visibility};
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+use crate::{FixAvailability, Violation};
+
+/// ## What it does
+/// Detects class `@property` methods that does not have a `return` statement.
+///
+/// ## Why is this bad?
+/// Property methods are expected to return a computed value, a missing return in a property usually indicates an implementation mistake.
+///
+/// ## Example
+/// ```python
+/// class User:
+///     @property
+///     def full_name(self):
+///         f"{self.first_name} {self.last_name}"
+/// ```
+///
+/// Use instead:
+/// ```python
+/// class User:
+///     @property
+///     def full_name(self):
+///         return f"{self.first_name} {self.last_name}"
+/// ```
+///
+/// ## References
+/// - [Python documentation: The property class](https://docs.python.org/3/library/functions.html#property)
+#[derive(ViolationMetadata)]
+#[violation_metadata(preview_since = "0.14.7")]
+pub(crate) struct PropertyWithoutReturn {
+    name: String,
+}
+
+impl Violation for PropertyWithoutReturn {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::None;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let Self { name } = self;
+        format!("{name} is a property without a `return` statement")
+    }
+}
+
+/// RUF069
+pub(crate) fn property_without_return(checker: &Checker, function_def: &StmtFunctionDef) {
+    let semantic = checker.semantic();
+
+    if checker.source_type.is_stub() || semantic.in_protocol_or_abstract_method() {
+        return;
+    }
+
+    let StmtFunctionDef {
+        decorator_list,
+        body,
+        name,
+        ..
+    } = function_def;
+
+    if !visibility::is_property(decorator_list, [], semantic)
+        || visibility::is_overload(decorator_list, semantic)
+        || function_type::is_stub(function_def, semantic)
+    {
+        return;
+    }
+
+    let mut visitor = PropertyVisitor::default();
+    visitor.visit_body(body);
+    if visitor.found {
+        return;
+    }
+
+    checker.report_diagnostic(
+        PropertyWithoutReturn {
+            name: name.to_string(),
+        },
+        function_def.range(),
```

Comment:
Sorry to add one more change needed. We should pick a smaller range for the diagnostic, rather than throwing a yellow squiggle on an entire function in an editor, as changing the diagnostic range later will potentially be a [breaking change](https://github.com/astral-sh/ruff/issues/14900#issuecomment-2543923625).

I think I'd suggest just marking the `@property` decorator itself, since that's the part that's unfulfilled by the body of the function.